core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -44,26 +45,20 @@ public CorePriceService(PriceFeedService priceFeedService) {
     public void getMarketPrice(String currencyCode, Consumer<Double> resultHandler) {
         String upperCaseCurrencyCode = currencyCode.toUpperCase();
 
-        if (!isFiatCurrency(upperCaseCurrencyCode))
-            throw new IllegalStateException(format("%s is not a valid currency code", upperCaseCurrencyCode));
-
         if (!priceFeedService.hasPrices())
             throw new IllegalStateException("price feed service has no prices");
 
-        try {
-            priceFeedService.setCurrencyCode(upperCaseCurrencyCode);
-        } catch (Throwable throwable) {
-            log.warn("Could not set currency code in PriceFeedService", throwable);
+        // fetch fresh prices if the cache is older than 2 minutes
+        if (System.currentTimeMillis() - priceFeedService.getLastRequestTimeStamp().getTime() > TimeUnit.MINUTES.toMillis(2)) {
```

Comment:
If I understand right, the API can serve a stale price for almost 2 minutes with this change. Also this approach merely avoids making a simultaneous request at the API level, whereas solving the underlying issue is preferred.